### ISSUE TITLE
Unactionable warning when booting Hibernate Reactive

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#45263]
When running the Hibernate Reactive ITs in `integration-tests/hibernate-reactive-postgresql`, I get the following warning:

> 2024-12-23 20:12:48,962 WARN  [io.qua.hib.orm.run.ser.QuarkusRuntimeInitDialectFactory] (JPA Startup Thread) Persistence unit default-reactive: Could not retrieve the database version to check it is at least 12.0.0

Two questions:

- I'm not sure this is normal as I don't have the same warning with Hibernate ORM in `integration-tests/jpa-h2`.
- Should we improve the warning to make it more actionable? I think in this case, we should probably set the version ourselves in the configuration?

**Comment by yrodiere** [comment]
> I'm not sure this is normal as I don't have the same warning with Hibernate ORM in `integration-tests/jpa-h2`.

This typically happens when we're not able to reach the DB on startup, and I don't think we have such use cases for ORM.

> I think in this case, we should probably set the version ourselves in the configuration?

I don't think so, since the version we're not able to retrieve is the actual version of the DB we're connecting to. No way to hardcode that.

> Should we improve the warning to make it more actionable?

We could add a hint such as "is the database reachable?", but beyond that there's not much we can do. This warning happens either if the DB is unreachable on startup (which is technically [not supported yet](https://github.com/quarkusio/quarkus/pull/43396)) or if there is a bug on our side.

In this case I wonder if Hibernate Reactive just doesn't have the same version detection code that ORM has, at least when the datasource isn't JDBC-enabled. Because the version detection code in ORM is purely JDBC-based.
Or we (I) simply didn't wire the detection properly for Hibernate Reactive in Quarkus.
Maybe @DavideD can have a look?

FWIW the detection happens here in Quarkus: https://github.com/quarkusio/quarkus/blob/6daa2ea1952c0326afd232f453d378d01296e268/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/QuarkusRuntimeInitDialectFactory.java#L101-L117

And is fed information generated by this code in Hibernate ORM: https://github.com/hibernate/hibernate-orm/blob/748fa8f7e1cafae4d60618d9082e3523ae3cf7b2/hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java#L162-L171

Note that we'll change that code a bit once we upgrade to ORM7, so that the dialect itself is in charge of detecting the version (which it can do more finely): https://github.com/quarkusio/quarkus/pull/43764/files#diff-86d161403825b101dd6e16f36ea3588690195d15079b74becd72409f3ab173a5L109-R109

**Comment by DavideD** [comment]
> In this case I wonder if Hibernate Reactive just doesn't have the same version detection code that ORM has

Yes, Hibernate Reactive has it's own way to detect the database: https://github.com/hibernate/hibernate-reactive/blob/37d57f6429fc20c5cc1e30f5fc22c6cde343d714/hibernate-reactive-core/src/main/java/org/hibernate/reactive/provider/service/NoJdbcEnvironmentInitiator.java#L81

We basically recreate the dialect but override the resolution info.

Quarkus seems to skip some steps when using Hibernate Reactive (it makes sense because we don't use JDBC) and `QuarkusRuntimeInitDialectFactory#retrieveDbVersion` is never called.

> Note that we'll change that code a bit once we upgrade to ORM7, so that the dialect itself is in charge of detecting the version 

Let's try not to forget about Hibernate Reactive when we do this.

**Comment by yrodiere** [comment]
> > Note that we'll change that code a bit once we upgrade to ORM7, so that the dialect itself is in charge of detecting the version
> 
> Let's try not to forget about Hibernate Reactive when we do this.

Can you please add a rejecting review on my PR mentioning that, so I don't forget?

https://github.com/quarkusio/quarkus/issues/45263

**Comment by gsmet** [comment]
> Quarkus seems to skip some steps when using Hibernate Reactive (it makes sense because we don't use JDBC) and QuarkusRuntimeInitDialectFactory#retrieveDbVersion is never called.

But doesn't the warning come from this very code?

FWIW, I'm running the tests with a container started so I would expect them to be able to contact the database.

**Comment by DavideD** [comment]
No, the warning comes from [QuarkusRuntimeInitDialectFactory#buildDialect](https://github.com/quarkusio/quarkus/blob/6daa2ea1952c0326afd232f453d378d01296e268/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/QuarkusRuntimeInitDialectFactory.java#L67), and the log happens because `retrieveDbVersion` has not being called before (and therefore the actual db version has not been initialized).

**Comment by mahieddine** [comment]
I had the same issue, the only action I found to get rid of it was to add the following in my application.properties file

`quarkus.hibernate-orm.dialect=org.hibernate.dialect.PostgreSQLDialect
quarkus.datasource.db-version=16.0`

Not ideal especially for the version, is this normal knowing that in my case the database already exists ?

**Comment by zbhavyai** [comment]
> I had the same issue, the only action I found to get rid of it was to add the following in my application.properties file
> 
> ```
> quarkus.hibernate-orm.dialect=org.hibernate.dialect.PostgreSQLDialect  
> quarkus.datasource.db-version=16.0
> ```
> 
> Not ideal especially for the version, is this normal knowing that in my case the database already exists and postgres is up&running when the application starts?

Thanks, this fixed my warning!
